File: pointer/#392_is_subsequence/isSubsequence.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function (s, t) {
  if (s.length > t.length) return false

  let ptr1 = 0, ptr2 = 0

  while (ptr1 < s.length && ptr2 < t.length) {
    if (s[ptr1] == t[ptr2]) {
      ptr1++
    }
    ptr2++
  }

  console.log(ptr1 === s.length);
  return ptr1 === s.length
};

isSubsequence("abc", "ahbgdc") // true
isSubsequence("axc", "ahbgdc") // false

File: pointer/#392_is_subsequence/isSubsequence.go
package main

func main() {
	isSubsequence("abc", "ahbgdc") // true
	isSubsequence("axc", "ahbgdc") // false
}

func isSubsequence(s string, t string) bool {
	if len(s) > len(t) {
		return false
	}

	p1, p2 := 0, 0

	for p1 < len(s) && p2 < len(t) {
		if s[p1] == t[p2] {
			p1++
		}
		p2++
	}

	return p1 == len(s)
}
